agent/util/subtree_unix.go

```diff
@@ -0,0 +1,224 @@
+// +build darwin linux
+
+package util
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+)
+
+const (
+	cleanupCheckAttempts   = 10
+	cleanupCheckTimeoutMin = 100 * time.Millisecond
+	cleanupCheckTimeoutMax = 1 * time.Second
+	contextTimeout         = 10 * time.Second
```

Comment:
I'm not sure what you mean. Do you have a case in mind that would need more time?
Separately, I extended it to 30 seconds because I thought with a maximum of one second we might exceed a ten second context before we've retried 10 times.

Reply:
I was mostly asking if `ps` always returns within 30 seconds. For example, can it spuriously context timeout on really slow machines or should we assume `ps` should finish within a few seconds?

Reply:
it seems to me we'll need to pick some duration after which we conclude ps is "stuck". I think you're right, though: we should pick a large duration where the probability that ps is really stuck is high. Maybe I'll go with 5 minutes.